src/Runtime/RedispatchHook.php

```diff
@@ -96,8 +97,11 @@ public function redispatch(InputInterface $input)
 
         $aliasRecord = $this->siteAliasManager()->getSelf();
         $process = $this->processManager->drushSiteProcess($aliasRecord, $redispatchArgs, $redispatchOptions);
-        $process->setTty($this->getConfig()->get('ssh.tty', $input->isInteractive()));
-        $process->setInput(STDIN);
+        if (!TerminalUtils::stdinIsTerminal()) {
```

Comment:
I am tempted to simplify this entire block to:
```
$process->setDefaultRedirectBehavior(STDIN, $input, $config, 'ssh.tty');
```
That's too many parameters, though.

Maybe:
```
$process->setDefaultRedirectBehavior(STDIN, $input, $this->getConfig()->get('ssh.tty'));
```